backend/provisioner/deployment/deployment.go

```diff
@@ -0,0 +1,133 @@
+package deployment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1beta1/provisioner"
+)
+
+type TaskState string
+
+const (
+	TaskStatePending TaskState = ""
+	TaskStateRunning TaskState = "running"
+	TaskStateDone    TaskState = "done"
+	TaskStateFailed  TaskState = "failed"
+)
+
+// Task is a unit of work for a deployment
+type Task struct {
+	Handler  Provisioner
+	Module   string
+	State    TaskState
+	Desired  []*provisioner.Resource
+	Existing []*provisioner.Resource
+	// populated only when the task is done
+	Output []*provisioner.Resource
```

Comment:
I think it would be cleaner if this wasn't public, and was instead returned by `Progress()` once the task is complete?